src/FSharp.Analyzers.Cli/Program.fs

```diff
@@ -0,0 +1,186 @@
+                []
+
+        return filesToCheck
+    } |> Async.RunSynchronously
+
+let typeCheckFile (file,opts) =
+    let text = File.ReadAllText file
+    let st = SourceText.ofString text
+    let (parseRes, checkAnswer) =
+        checker.ParseAndCheckFileInProject(file, 1, st, opts)
+        |> Async.RunSynchronously
+
+    match checkAnswer with
+    | FSharpCheckFileAnswer.Aborted ->
+        printError "Checking of file %s aborted" file
+        None
+    | FSharpCheckFileAnswer.Succeeded(c) ->
+        Some (file, text, parseRes, c)
+
+let createContext (file, text: string, p: FSharpParseFileResults,c: FSharpCheckFileResults) =
+    match p.ParseTree, c.ImplementationFile with
+    | Some pt, Some tast ->
+        let context : Context = {
+            FileName = file
+            Content = text.Split([|'\n'|])
+            ParseTree = pt
+            TypedTree = tast
+            Symbols = c.PartialAssemblySignature.Entities |> Seq.toList
+        }
+        Some context
+    | _ -> None
+
+let runProject proj (globs: Glob list) analyzers  =
+    let path =
+        Path.Combine(Environment.CurrentDirectory, proj)
+        |> Path.GetFullPath
+
+    let files =
+        loadProject path
+        |> List.filter (fun (f,_) ->
+            match globs |> List.tryFind (fun g -> g.IsMatch f) with
+            | None -> true
+            | Some g ->
+                printInfo "Ignoring file %s for pattern %s" f g.Pattern
+                false)
+        |> List.choose typeCheckFile
+        |> List.choose createContext
+
+
+    files
+    |> Seq.collect (fun ctx ->
+        printInfo "Running analyzers for %s" ctx.FileName
+        analyzers |> Seq.collect (fun analyzer -> analyzer ctx)
+    )
+    |> Seq.toList
+
+let printMessages failOnWarnings (msgs: Message list) =
+    if verbose then printfn ""
+
+    msgs
+    |> Seq.iter(fun m ->
+        let color =
+            match m.Severity with
+            | Error -> ConsoleColor.Red
+            | Warning when failOnWarnings |> List.contains m.Code -> ConsoleColor.Red
+            | Warning -> ConsoleColor.DarkYellow
+            | Info -> ConsoleColor.Blue
+
+        Console.ForegroundColor <- color
+        printfn "%s(%d,%d): %s %s - %s" m.Range.FileName m.Range.StartColumn m.Range.StartLine (m.Severity.ToString()) m.Code m.Message
+        Console.ForegroundColor <- ConsoleColor.White
+    )
+    msgs
+
+let calculateExitCode failOnWarnings (msgs: Message list): int =
+    let check =
+        msgs
+        |> List.exists (fun n -> n.Severity = Error || (n.Severity = Warning && failOnWarnings |> List.contains n.Code) )
+
+    if check then -12345 else 0
+
+[<EntryPoint>]
+let main argv =
+    let results = parser.ParseCommandLine argv
+    verbose <- results.Contains <@ Verbose @>
+    printInfo "Running in verbose mode%s" ""
+
+    let failOnWarnings = results.GetResult(<@ Fail_On_Warnings @>, [])
+    printInfo "Fail On Warnings: %A" failOnWarnings
```

Comment:
Depending on how many warnings `%A` might not print all of them.  I think it's worth considering a `String.join` here.